homeassistant/components/fritzbox/climate.py

```diff
@@ -204,6 +201,7 @@ def preset_mode(self) -> str | None:
 
     async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set preset mode."""
+        self.check_active_or_lock_mode()
         if self.data.holiday_active or self.data.summer_active:
```

Comment:
this condition is not needed anymore and can be removed

Reply:
no no ... the `if self.data.holiday_active or self.data.summer_active:` block was meant, not the `self.check_active_or_lock_mode()` :wink: